### ISSUE TITLE
[FEATURE] Add API to get count of users matching research profile

### ISSUE THREAD

**Issue by shurwit** [rokwire/groups-building-block#291]
**Is your feature request related to a problem? Please describe.**
Currently a researcher creating a research group will not know how many users match the criteria they specified and may end up with a candidate pool that is too small.

**Describe the solution you'd like**
We should expose a `POST /api/research-profile/user-count` API that allows a researcher to retrieve a count of the number of users that match their currently selected set of demographic criteria. The body will contain a `map[string]map[string][]string` of the currently selected `ResearchProfile` for the group. Requests to this API should require the `research_group_admin` permission.

The API should convert this into the appropriate format and send a request to the Core BB API introduced in https://github.com/rokwire/core-building-block/issues/619.  Note that this issue will depend on #290 since we will need to use a service account to call the Core BB API.

**Describe alternatives you've considered**
We could expose the count API directly on the Core BB, however this would either require making an API specific to research groups/research demographic questions on the Core BB, or exposing access to additional properties of the account not needed for this purpose.
